src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Events/PluginSourceErrorEventArgs.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Events
+{
+    /// <summary>
+    ///    Base event arguments type for when a plugin source encounters an error.
+    /// </summary>
+    public abstract class PluginSourceErrorEventArgs
+        : EventArgs
+    {
+        protected PluginSourceErrorEventArgs(PluginSource pluginSource)
+            : this(pluginSource, $"Error occurred when interacting with plugin source: {pluginSource.Uri}")
+        {
+        }
+
+        protected PluginSourceErrorEventArgs(PluginSource pluginSource, string errorMessage)
```

Comment:
nit: may want to have an optional `ExceptionDispatchInfo` on here

Reply:
Hmm, I see the other class. I would prefer just having one sealed class that has an `Exception` property that may be null